docs/core-concepts/console/step-by-step-guide.md

```diff
@@ -0,0 +1,643 @@
+
+If you're unfamiliar with the .NET CLI toolset, read [the overview](../core-sdk/sdk-overview.md).  It's also helpful to have an understanding of the [console app paradigm](paradigm.md) and how [native binary compilation](single-binaries.md) works.
+
+## Prerequisites
+
+Before you begin, ensure you have the [latest .NET CLI tooling](http://dotnet.github.io/getting-started/).  You'll also need a text editor.
+
+## Hello, Console App!
+
+First, navigate to or create a new folder with a name you like.  "Hello" is the name chosen for the sample code, which can be found [here](https://github.com/dotnet/core-docs/samples/core-projects/console-apps/Hello).
+
+Open up a command prompt and type the following:
+
+```
+$ dotnet new
+$ dotnet restore
+$ dotnet run
+```
+
+Let's do a quick walkthrough:
+
+1. `$ dotnet new`
+
+   `dotnet new` creates an up-to-date `project.json` file with NuGet dependencies necessary to build a console app.  It also creates a `Program.cs`, a basic file containing the entry point for the application.
+   
+   `project.json`:
+   ```javascript
+   {
+        "version": "1.0.0-*",
+        "buildOptions": {
+            "emitEntryPoint": true
+        },
+        "dependencies": {
+            "Microsoft.NETCore.App": {
+                "type": "platform",
+                "version": "1.0.0-rc2-3002702"
+            }
+        },   
+       "frameworks": {
+            "netcoreapp1.0": {
+                "imports": "dnxcore50"
+            }
+        }
+   }
+   ```
+   `Program.cs`:
+   ```csharp
+   using System;
+
+   namespace ConsoleApplication
+   {
+       public class Program
+       {
+           public static void Main(string[] args)
+           {
+               Console.WriteLine("Hello World!");
+           }
+       }
+   }
+   ```
+
+2. `$ dotnet restore`
+
+   `dotnet restore` analyzes the `project.json` file, downloads the dependencies stated in the file (or grabs them from a cache on your machine), and writes the `project.lock.json` file.  The `project.lock.json` file is necessary to be able to compile and run.
+   
+   The `project.lock.json` file is a persisted and complete set of the graph of NuGet dependencies and other information describing an app.  This file is read by other tools, such as `dotnet build` and `dotnet run`, enabling them to process the source code with a correct set of NuGet dependencies and binding resolutions.
+   
+3. `$ dotnet run`
+
+   `dotnet run` launches the runtime and executes the application created from the `project.json` file in the current directory. If necessary, it compiles the source to Intermediate Language (IL) and generates `Hello.dll` `.dll` file which contains the IL using the `dotnet build` command.
```

Comment:
Err, not really, it calls `dotnet build` and then `dotnet [assembly.dll]` to run it.